File: SmartTooling-API/Models/ProductionBP/PBP_Pad_Print_Setting.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SmartTooling_API.Models.ProductionBP
{
    public partial class PBP_Pad_Print_Setting
    {
        [Key]
        [StringLength(50)]
        public string factory_id { get; set; }
        [Key]
        [StringLength(8)]
        public string model_no { get; set; }
        [Key]
        [StringLength(100)]
        public string component_name { get; set; }
        [Key]
        [StringLength(10)]
        public string material_type_id { get; set; }
        [Key]
        [StringLength(100)]
        public string chemical_ink { get; set; }
        [Required]
        [StringLength(200)]
        public string material_description { get; set; }
        [Required]
        [StringLength(10)]
        public string pad_shape_id { get; set; }
        public int number_of_pad_hits { get; set; }
        [Required]
        [StringLength(10)]
        public string machine_vendor_id { get; set; }
        [Required]
        [StringLength(200)]
        public string machine_model { get; set; }
        public bool is_rotary_table_used { get; set; }
        [Required]
        [StringLength(200)]
        public string chemical_hardener { get; set; }
        [Required]
        [StringLength(200)]
        public string chemical_additive { get; set; }
        [Required]
        [StringLength(200)]
        public string chemical_primer { get; set; }
        [Required]
        [StringLength(200)]
        public string chemical_others { get; set; }
        public bool article_no_is_general { get; set; }
        [StringLength(500)]
        public string article_no_remarks { get; set; }
        [Required]
        [StringLength(400)]
        public string component_photo_url { get; set; }
        [Required]
        [StringLength(400)]
        public string operation_video_url { get; set; }
        [Required]
        [StringLength(50)]
        public string create_by { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime create_time { get; set; }
        [Required]
        [StringLength(50)]
        public string update_by { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime update_time { get; set; }
    }
}

File: SmartTooling-SPA/src/app/_core/_models/production-bp/pbp-pad-print-setting.ts
export interface PbpPadPrintSetting {
    factory_id: string;
    model_no: string;
    model_name: string;
    component_name: string;
    material_type_id: string;
    material_type_name: string;
    chemical_ink: string;
    dev_season: string;
    production_season: string;
    material_description: string;
    pad_shape_id: string;
    pad_shape_name: string;
    number_of_pad_hits: number;
    machine_vendor_id: string;
    machine_vendor_name: string;
    machine_model: string;
    is_rotary_table_used: boolean;
    is_rotary_table_used_result: string;
    chemical_hardener: string;
    chemical_additive: string;
    chemical_primer: string;
    chemical_others: string;
    article_no_is_general: boolean;
    article_no_is_general_result: string;
    article_no_remarks: string;
    component_photo_url: string;
    component_photo_url_result: string;
    operation_video_url: string;
    operation_video_url_result: string;
    component_photo_url_file: string;
    operation_video_url_file: string;
    create_by: string;
    create_time: string;
    update_by: string;
    update_time: string;
}